src/Symfony/Component/Security/Http/Authenticator/AccessToken/QueryAccessTokenExtractor.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Extracts a token from a query string parameter.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+final class QueryAccessTokenExtractor implements AccessTokenExtractorInterface
+{
+    public const PARAMETER = 'access_token';
+    private string $parameter;
+
+    public function __construct(string $parameter = self::PARAMETER)
+    {
+        $this->parameter = trim($parameter);
```

Comment:
Why do I need to trim here?

Reply:
In case of. But OK I removed all the unnecessary `trim`s